### ISSUE TITLE
Average spike rate: 0.0

### ISSUE THREAD

**Issue by dahexiangdongliu** [NeuromorphicProcessorProject/snn_toolbox#61]
Hello,

Thanks for your wonderful work, it helps me a lot to understand SNN.
I am trying to runing the example, when I run the mnist_keras_INI.py, the accuracy is

> Moving accuracy of SNN (top-1, top-1): 96.68%, 96.67%.
Moving accuracy of ANN (top-1, top-1): 96.94%, 99.93%.

But I get this

> Current accuracy of batch:
6.25%_12.50%_14.06%_18.75%_18.75%_31.25%_56.25%_70.31%_81.25%_90.62%_90.62%_92.19%_95.31%_96.88%_95.31%_96.88%_96.88%_96.88%_96.88%_98.44%_98.44%_98.44%_98.44%_98.44%_98.44%_98.44%_98.44%_98.44%_98.44%_98.44%_98.44%_98.44%_98.44%_98.44%_98.44%_98.44%_98.44%_98.44%_98.44%_98.44%_98.44%_98.44%_98.44%_98.44%_98.44%_98.44%_98.44%_98.44%_98.44%_98.44%_Average spike rate: 0.0 spikes per simulation time step.

Could you tell me  why there's no spike, but the accuracy is right? I only change the num_to_test to 10000 and batch_size to 64 in the configuration of mnist_keras_INI.py.

And I also have trained an ANN model myself on MNIST, then use the toolbox to convert and simulate, the result is similar.

I have used two version of the toolbox, snntoolbox 0.4.1 with keras2.2.4, tensorflow-gpu1.9.0, and snntoolbox 0.5.0 with tensorflow-gpu2.2.0. The result is similar too.

Thank you!

**Comment by rbodo** [comment]
Hi,

I can't reproduce the issue; when I run the example (also with your modifications), I get an average spike rate of 0.082...

I'm using tensorflow 2.2.0. I don't think it will make a difference, but you shouldn't explicitly have to install the -gpu version, that is contained in plain ``pip install tensorflow`` now. Also, keras does not need to be installed separately any more, the toolbox now uses keras from within tf. Maybe that solves it, otherwise you could try setting a breakpoint [here](https://github.com/NeuromorphicProcessorProject/snn_toolbox/blob/73d8f19730da9a6ca4b5a2055655c6efb026b644/snntoolbox/simulation/utils.py#L707) and check whether the spiketrains variable contains any nonzero values.

**Comment by dahexiangdongliu** [comment]
Hi,

Thanks for your reply! I have created a new environment and installed matplotlib, then the average spike rate was right. 

I have set a breakpoint where you recommended, and I found that the self.spiketrains_n_b_l_t was None when the matplotlib was not installed. 

When I have installed matplotlib, and removed the `config['output']` of the config of mnist_keras_INI.py, the self.spiketrains_n_b_l_t was also None.

Does it mean the spikes printed here is just for plotting, instead of the real spikes when the toolbox is simulating?

**Comment by rbodo** [comment]
That is correct. When running a large network, recording the spiketrains gets very time and memory consuming, so it is enabled only if needed for plotting. Of course the spikes are still there, we just don't write them out.

In any case, the toolbox shouldn't report a zero rate, I fixed this now.

Thanks for noticing.
